libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/NavigatorMapper.kt

```diff
@@ -78,39 +76,38 @@ internal class NavigatorMapper {
             val legProgressBuilder = RouteLegProgress.Builder()
             val stepProgressBuilder = RouteStepProgress.Builder()
 
-            ifNonNull(route.legs()) { legs ->
+            ifNonNull(route.legs(), activeGuidanceInfo) { legs, activeGuidanceInfo ->
                 var currentLeg: RouteLeg? = null
                 if (legIndex < legs.size) {
                     currentLeg = legs[legIndex]
                     legProgressBuilder.legIndex(legIndex)
                     legProgressBuilder.routeLeg(currentLeg)
 
-                    // todo mapbox java issue - leg distance is nullable
```

Comment:
cool this is fixed?

Reply:
distanceTraveled / (currentLeg.distance()?.toFloat() ?: 0f)

lol i guess it has to be since this is a divide by zero

Reply:
Yeah, as we're not relying on `currentLeg.distance()?.toFloat()` (that came from `mapbox-java`) anymore and NN one is `@NonNull` 🚀

Reply:
Operation is done internally now in NN and we use it directly so no need to anything from our side https://github.com/mapbox/mapbox-navigation-android/pull/3604#issuecomment-701503824

> also adds additional fields that may make some of the computations inside the SDK code (fractions traveled) unnecessary.